lambdas/call-ticf-cri/src/main/java/uk/gov/di/ipv/core/callticfcri/service/TicfCriService.java

```diff
@@ -0,0 +1,133 @@
+package uk.gov.di.ipv.core.callticfcri.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nimbusds.jwt.SignedJWT;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import software.amazon.lambda.powertools.tracing.Tracing;
+import uk.gov.di.ipv.core.callticfcri.dto.TicfCriDto;
+import uk.gov.di.ipv.core.callticfcri.exception.TicfCriServiceException;
+import uk.gov.di.ipv.core.library.dto.BackEndCriConfig;
+import uk.gov.di.ipv.core.library.exceptions.VerifiableCredentialException;
+import uk.gov.di.ipv.core.library.persistence.item.ClientOAuthSessionItem;
+import uk.gov.di.ipv.core.library.persistence.item.IpvSessionItem;
+import uk.gov.di.ipv.core.library.service.ConfigService;
+import uk.gov.di.ipv.core.library.verifiablecredential.validator.VerifiableCredentialJwtValidator;
+
+import java.io.IOException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static uk.gov.di.ipv.core.library.domain.CriConstants.TICF_CRI;
+
+public class TicfCriService {
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+    public static final String TRUSTMARK = "https://oidc.account.gov.uk/trustmark";
+    public static final String X_API_KEY_HEADER = "x-api-key";
+
+    private final ConfigService configService;
+    private final HttpClient httpClient;
+    private final VerifiableCredentialJwtValidator jwtValidator;
+
+    public TicfCriService(ConfigService configService) {
+        this.configService = configService;
+        this.httpClient = HttpClient.newHttpClient();
+        this.jwtValidator = new VerifiableCredentialJwtValidator(configService);
+    }
+
+    protected TicfCriService(
+            ConfigService configService,
+            HttpClient httpClient,
+            VerifiableCredentialJwtValidator jwtValidator) {
+        this.configService = configService;
+        this.httpClient = httpClient;
+        this.jwtValidator = jwtValidator;
+    }
+
+    public List<SignedJWT> getTicfVc(
+            ClientOAuthSessionItem clientOAuthSessionItem,
+            IpvSessionItem ipvSessionItem,
+            List<String> credentials)
+            throws TicfCriServiceException {
+        try {
+            BackEndCriConfig ticfCriConfig = configService.getBackEndCriConfig(TICF_CRI);
```

Comment:
Question about our code style - I notice you don't use `var` very much?

I'd probably lean towards using it for pretty much everything, unless there's a need to cast to a wider type, but keen to hear arguments. (We should probably also _try_ and be consistent - although I know at the moment we aren't at all!)

Reply:
I'm not a fan of using `var`. I much prefer being able to immediately see what type something is. When var is used I'll often have to go find a method definition to determine what type something is. I'll _occasionally_ use it if a type is some obnoxiously long generic type of something. But I usually don't find an issue with being explicit and more informative.

Reply:
Ok interesting - I mostly rely on my IDE to tell me the type of variables (which it can still do with `var`) and in my opinion makes it cleaner to write, but it's definitely a preference. It's already a mix so won't try and do anything about this now, but at some point I might try to write a style-guide for us.

(I guess also as an observation, a lot of modern languages have shifted towards var-esque syntax!)

Reply:
Maybe my IDE isn’t set up right 😅. But if I’m reviewing a PR in GitHub, I don’t really want to have to check out the PR into my IDE. Although GitHub sides have some fancy new features - perhaps it can tell me.